benchmarks/distributed/rpc/parameter_server/trainer/trainer.py

```diff
@@ -175,3 +177,91 @@ def clear_metrics(self):
         A method that clears __metrics_logger recorded metrics.
         """
         return self.__metrics_logger.clear_metrics()
+
+
+class DdpTrainer(TrainerBase):
+
+    def __init__(
+        self,
+        rank,
+        trainer_count,
+        process_group,
+        use_cuda_rpc,
+        server_rref,
+        backend,
+        epochs,
+        preprocess_data,
+        create_criterion,
+        create_ddp_model,
+        hook_state_class,
+        hook,
+        iteration_step
+    ):
+        r"""
+        A trainer that implements a DDP training algorithm using a simple hook that performs allreduce
+        using the process_group implementation.
+        Args:
+            rank (int): worker rank
+            trainer_count (int): count of trainer in the world
```

Comment:
can rank and trainer_count be derived from `process_group`?

Reply:
Yes.